src/main/java/com/ratifire/devrate/service/user/UserService.java

```diff
@@ -899,4 +905,21 @@ public void sendTestNotification(long userId, NotificationDto notificationDto) {
         .build();
     notificationService.sendTestNotification(userEmail, notification);
   }
+
+  /**
+   * Adds feedback for a user if they are eligible, otherwise throws exception.
+   *
+   * @param userId      the ID of the user
+   * @param feedbackDto the feedback details to add
+   */
+  public void addFeedback(long userId, FeedbackDto feedbackDto) {
+    User user = findUserById(userId);
+    if (feedbackService.isUserAbleToSendFeedback(user)) {
+      Feedback entity = feedbackMapper.toEntity(feedbackDto);
+      entity.setUser(user);
+      feedbackService.save(entity);
+    } else {
+      throw new FeedbackFrequencyLimitReached(user.getId());
+    }
```

Comment:
this method should do only 1 thing and as the name suggests it needs to add feedback and not check the frequency limit. Why this method is not in the feedback service?